opinions/spring_boot_opinions.go

```diff
@@ -310,18 +302,27 @@ type SpringBootBOMMetadataDependency struct {
 
 type springApplicationPropertiesKey struct{}
 
-func StashSpringApplicationProperties(ctx context.Context, props map[string]string) context.Context {
+type SpringApplicationProperties map[string]string
```

Comment:
It would be good to generate this struct as there is a well defined schema of all exposed Spring configuration properties (probably a separate issue)

Reply:
https://github.com/spring-projects/spring-boot/blob/master/spring-boot-project/spring-boot-autoconfigure/src/main/resources/META-INF/additional-spring-configuration-metadata.json

Reply:
`map[string]string` faithfully models a properties file. It also has the advantage of not being dependent on any particular version of boot. More importantly it doesn't limit third party configuration properties.